### PR TITLE
Fixing test module partitioning_warnings

### PR DESCRIPTION
**Pull request by SergioAtSUSE**
## Motivation

- Related job: https://openqa.suse.de/tests/1509521
- Related ticket: https://progress.opensuse.org/issues/32527
- Verification run: http://copland.arch.suse.de/tests/939#step/partitioning_warnings/33

----

## Remove invalid soft-failure
Manually verified partition role with DOS partition table.

- Related ticket: https://bugzilla.suse.com/show_bug.cgi?id=1055743
- superseeds: https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/4531

----

## Add BIOS boot partition

For this test suite it was not added. x86_64 on GTP table needs this partition.

### CODE REVIEW

**Comment by rwx788** [comment]
@SergioAtSUSE that's a good point, I guess we can exit after starting the installation. So is this PR WIP then?

**Comment by SergioAtSUSE** [comment]
Verification run: http://copland.arch.suse.de/tests/963#step/partitioning_warnings/38

File: tests/installation/partitioning_warnings.pm [hunk 1]
```diff
@@ -50,7 +49,7 @@ sub run {
         assert_screen 'partition-warning-no-efi-boot';
         wait_screen_change { send_key 'alt-y' };    # yes
     }
-    if (is_storage_ng && !check_screen('partitioning-edit-proposal-button', 0)) {
+    if (is_storage_ng && check_screen('partitioning-edit-proposal-button', 0)) {
```
Comment (rwx788) [inline tests/installation/partitioning_warnings.pm]: We should get rid of this, as bug is resolved and there will be no warning for missing swap, as we have new checkbox for that. So this change doesn't make sense anymore...

**Comment by SergioAtSUSE** [comment]
New verification run: http://copland.arch.suse.de/tests/966#step/partitioning_warnings/38

Expected to fail there. The workaround should be removed with: https://progress.opensuse.org/issues/31066

**Comment by SergioAtSUSE** [comment]
New verification run: http://copland.arch.suse.de/tests/969#step/partitioning_warnings/37

**Comment by okurz** [comment]
+1, ready to be merged?

**Pull request merged**